pkg/registry/apiserverinternal/storageversion/strategy.go

```diff
@@ -43,10 +44,17 @@ func (storageVersionStrategy) NamespaceScoped() bool {
 
 // PrepareForCreate clears the status of an StorageVersion before creation.
 func (storageVersionStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
+	sv := obj.(*apiserverinternal.StorageVersion)
+	sv.Generation = 1
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (storageVersionStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Object) {
+	newStorageVersion := obj.(*apiserverinternal.StorageVersion)
+	oldStorageVersion := old.(*apiserverinternal.StorageVersion)
+	if !apiequality.Semantic.DeepEqual(newStorageVersion.Status, oldStorageVersion.Status) {
+		newStorageVersion.Generation = oldStorageVersion.Generation + 1
```

Comment:
We usually update generation when there is a change to the object's spec, but storageVersion doesn't have a spec at all.